### PR TITLE
Migrate site to new setup (Hugo + Netlify) (work in progress)

### PR DESCRIPTION
**Pull request by lucperkins**
An up-to-date preview can be seen here: https://deploy-preview-25--jaegertracing.netlify.com/

Criteria for merging this PR:

- [x] The aesthetic of the new website is sufficiently pleasing and polished to meet the standards of core project maintainers
- [x] All information previously available on the [Read the Docs](http://jaeger.readthedocs.io/en/latest/) site is available on the new site
- [x]  All existing links for the current site will be properly aliased (preventing unwanted 404s from Google searches

Please note that this PR provides virtually no changes in content from the original [Read the Docs](http://jaeger.readthedocs.io/en/latest/) site. Content changes will come in future PRs.

### CODE REVIEW

File: config.yaml [hunk 1]
```diff
@@ -10,8 +10,7 @@ disableKinds:
 - taxonomyTerm
 
 params:
-  description: |
-    Jaeger, inspired by [Dapper](https://research.google.com/pubs/pub36356.html) and [OpenZipkin](https://zipkin.io/), is a distributed tracing system released as open source by Uber Technologies. It can be used for monitoring microservice-based architectures.
+  tagline: Monitor and troubleshoot transactions in distributed systems
```
Comment (jpkrohling) [inline config.yaml]: I like the new version (`Monitor....`) more than the old one (`... inspired by...`), but how about something like: 

> Open Source distributed tracing system created to monitor cloud native applications and microservices architectures

Credit goes to @JStickler and @objectiser  :)

Comment (yurishkuro) [reply config.yaml]: I think the difference between the Title and the Tagline is that the Title is "what is it" (end-to-end distributed tracing) and the Tagline is "what is it for". So let's take this apart:
  * "distributed tracing system" - redundant as we already say it in the title
  * "open source" - maybe, but we could also say that in the following section, not in the tagline (cf. https://kubernetes.io/)
  * "microservices architectures" - a misnomer, microservices are not an architecture, they are a deployment choice. The same architecture can be implemented as a monolith or as microservices
  * technically Jaeger can even be used to profile monoliths (in fact that's how tracing started at Uber), but I an not sure we need to mention that
  * both "cloud native" and "microservices" serve more as buzz words than an accurate description of the target. "complex distributed systems" would be more accurate. But buzz words might be good for PR

Possible alternative (two lines): 
```
Monitor and troubleshoot transactions in complex distributed systems,
cloud native and microservices-based application
```

Comment (yurishkuro) [reply config.yaml]: PS - I like the one-liner better

Comment (jpkrohling) [reply config.yaml]: "complex distributed systems" is pretty much a synonym for "microservices" nowadays :) 

Not relevant for this specific PR, but I would be interested in hearing more details about "microservices architecture" being a misnomer, as I have accepted Martin Fowler's definition so far, where he says that it's an "architectural style".

Comment (yurishkuro) [reply config.yaml]: > "complex distributed systems" is pretty much a synonym for "microservices" nowadays :)

yes and no. Do you consider Cassandra a microservices-based solution? I think most people wouldn't, but it's certainly falls under the umbrella of a "complex distributed system".

> more details about "microservices architecture" being a misnomer

I think on a high level "architecture" refers to logical components that work together as a single application. More than 10yrs ago, when nobody has used the term "microservices", we used to architect a system as a collection of services (SOA), each implementing a certain API and performing a certain business function. Then at deployment time we could decide whether we want to deploy several services in a single JVM (and let them talk via direct API calls) or in multiple JVMs (and let them talk via SOAP). Or take today's example - Jaeger at Uber. Our collector certainly qualifies as a microservice, yet it contains several business "services". Deciding whether a given service runs as a separate process or is running in a larger process that combines multiple services is a deployment question, not an architectural question.

Anyway, this is just an opinion.

**Comment by yurishkuro** [comment]
@lucperkins could you please outline the criteria for merging this PR? Looking at the notes from our email exchange it seems the only outstanding question is what we do with the existing readthedocs site.

**Comment by lucperkins** [comment]
@yurishkuro The criteria checklist for the PR is now up to date

**Comment by yurishkuro** [comment]
I just pushed the existing `master` to `readthedocs` branch, and changed readthedocs project config to build from that branch instead of master (https://readthedocs.org/projects/jaeger/builds/6959733/)

**Comment by lucperkins** [comment]
@yurishkuro I bumped the Hugo version again, as 0.36.1 is not available via Homebrew, which will make the macOS setup command (`make macos-setup`) fail.

**Comment by yurishkuro** [comment]
> I bumped the Hugo version again, as 0.36.1 is not available via Homebrew,

for me it was the other way, Makefile had 0.36 and it was failing on my Mac.

**Comment by lucperkins** [comment]
@yurishkuro Huh, weird. I just verified locally that 0.37.1 works, so we should be fine here. If not, please let me know ASAP.

**Comment by yurishkuro** [comment]
Neither 0.36 not 0.37.1 work for me, only 0.36.1. It's possible it's due to Uber running a local copy of brew repository (I see `brew update` contacting an internal site).
```
$ make macos-setup
brew switch hugo 0.37.1 && brew link --overwrite hugo
Error: hugo does not have a version "0.37.1" in the Cellar.
Versions available: 0.36.1
make: *** [macos-setup] Error 3
```

**Comment by yurishkuro** [comment]
do we actually need to pin the exact version? are we using some bleeding edge features? or perhaps brew can be told just the min required version?

**Pull request merged**